pkg/cmd/release/download/download.go

```diff
@@ -78,10 +90,29 @@ func NewCmdDownload(f *cmdutil.Factory, runF func(*DownloadOptions) error) *cobr
 
 	cmd.Flags().StringVarP(&opts.Destination, "dir", "D", ".", "The directory to download files into")
 	cmd.Flags().StringArrayVarP(&opts.FilePatterns, "pattern", "p", nil, "Download only assets that match a glob pattern")
+	cmd.Flags().StringVarP(&opts.ArchiveType, "archive", "A", "", "Download default assset with specified type (zip or tgz)")
 
 	return cmd
 }
 
+func checkArchiveTypeOption(opts *DownloadOptions) error {
+	if len(opts.ArchiveType) == 0 {
+		return nil
+	}
+
+	if len(opts.FilePatterns) > 0 {
+		return &cmdutil.FlagError{Err: errors.New("the '--pattern' and '--archive' flags cannot be set simultaneously")}
+	}
+	validArchiveTypes := map[string]struct{}{
+		"zip": {},
+		"tgz": {},
+	}
+	if _, ok := validArchiveTypes[opts.ArchiveType]; !ok {
```

Comment:
This is neat and pretty idiomatic Go, but we're unlikely to ever add more formats to this list, so let's keep it a simple conditional to avoid allocating a map:
```suggestion
	if opts.ArchiveType != "zip" && opts.ArchiveType != "tar.gz" {
```